lib/ain-evm/src/evm.rs

```diff
@@ -58,32 +67,34 @@ impl EVMServices {
     ///
     /// Returns an instance of the struct, either restored from storage or created from a JSON file.
     pub fn new() -> Result<Self, anyhow::Error> {
-        if let Some(path) = ain_cpp_imports::get_state_input_json() {
+        let services = if let Some(path) = ain_cpp_imports::get_state_input_json() {
             if ain_cpp_imports::get_network() != "regtest" {
                 return Err(anyhow!(
                     "Loading a genesis from JSON file is restricted to regtest network"
                 ));
             }
             let storage = Arc::new(Storage::new());
-            Ok(Self {
+            Self {
                 core: EVMCoreService::new_from_json(Arc::clone(&storage), PathBuf::from(path)),
                 block: BlockService::new(Arc::clone(&storage)),
                 receipt: ReceiptService::new(Arc::clone(&storage)),
                 logs: LogService::new(Arc::clone(&storage)),
                 filters: FilterService::new(),
                 storage,
-            })
+            }
         } else {
             let storage = Arc::new(Storage::restore());
-            Ok(Self {
+            Self {
                 core: EVMCoreService::restore(Arc::clone(&storage)),
                 block: BlockService::new(Arc::clone(&storage)),
                 receipt: ReceiptService::new(Arc::clone(&storage)),
                 logs: LogService::new(Arc::clone(&storage)),
                 filters: FilterService::new(),
                 storage,
-            })
-        }
+            }
+        };
+
+        Ok(services)
```

Comment:
Is there a rationale for changing the instantiation function to this?

Reply:
Old change I forgot to undo, will revert this.